### PR TITLE
Enable GenerateDocumentationFile. Add/fix XML docs.

### LINKED ISSUE #558: Add XML doc files to Blazor packages
We need the XML doc files in the Blazor packages for intellisense and for API ref doc generation.

### PR DESCRIPTION
**Pull request by SteveSandersonMS**
Fixes #558

### CODE REVIEW

File: src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorCodeDocItems.cs [hunk 1]
```diff
@@ -1,11 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-namespace Microsoft.AspNetCore.Blazor.Razor
-{
-    public static class BlazorCodeDocItems
```
Comment (SteveSandersonMS) [inline src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorCodeDocItems.cs]: This was not being used (AFAICT).

File: src/Microsoft.AspNetCore.Blazor/UIEventHandler.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using Microsoft.AspNetCore.Blazor.RenderTree;
```
Comment (SteveSandersonMS) [inline src/Microsoft.AspNetCore.Blazor/UIEventHandler.cs]: Needed because some of the `<see ...>` tags rely on it.

**Comment by SteveSandersonMS** [comment]
Merged so we can move towards a final RC build, but @rynowak @javiercn please do leave review comments if you have any or if you have concerns about including this.

File: src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorCodeDocItems.cs (hunk 1 above)
Comment (rynowak) [reply src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorCodeDocItems.cs]: 👍

File: src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComponentDocumentClassifierPass.cs [hunk 3]
```diff
@@ -11,32 +11,48 @@
 
 namespace Microsoft.AspNetCore.Blazor.Razor
 {
+    /// <summary>
+    /// A <see cref="DocumentClassifierPassBase"/> that recognizes Blazor components.
+    /// </summary>
     public class ComponentDocumentClassifierPass : DocumentClassifierPassBase, IRazorDocumentClassifierPass
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComponentDocumentClassifierPass.cs]: This should be internal instead, there's not scenario for user code to interact with this

File: src/Microsoft.AspNetCore.Blazor/Routing/NavLink.cs [hunk 4]
```diff
@@ -73,6 +73,7 @@ public void SetParameters(ParameterCollection parameters)
             _renderHandle.Render(Render);
         }
 
+        /// <inheritdoc />
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Blazor/Routing/NavLink.cs]: Note that this doesn't actually work (I don't think). Our docs stuff doesn't actually honour this

**Pull request merged**